src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/TupleExpansion.cs

```diff
@@ -0,0 +1,266 @@
+                    parentFullName,
+                    declaringType: field.DeclaringTypeAndInfo.ClrType,
+                    declaringTypeInfo: null,
+                    memberName: fieldName,
+                    memberAccessRequiresExplicitCast: false,
+                    memberIsStatic: false);
+            return value.GetFieldValue(fieldName, inspectionContext);
+        }
+
+        private sealed class Field
+        {
+            internal readonly TypeAndCustomInfo DeclaringTypeAndInfo;
+            internal readonly TypeAndCustomInfo FieldTypeAndInfo;
+            internal readonly FieldInfo FieldInfo; // type field
+            internal readonly string Name;
+            internal readonly Field Parent; // parent Rest field, if any
+
+            internal Field(
+                TypeAndCustomInfo declaringTypeAndInfo,
+                TypeAndCustomInfo fieldTypeAndInfo,
+                FieldInfo fieldInfo,
+                string name,
+                Field parent)
+            {
+                Debug.Assert(declaringTypeAndInfo.ClrType != null);
+                Debug.Assert(fieldTypeAndInfo.ClrType != null);
+                Debug.Assert(fieldInfo != null);
+                Debug.Assert(name != null);
+                Debug.Assert(declaringTypeAndInfo.Type.Equals(fieldInfo.DeclaringType));
+                Debug.Assert(fieldTypeAndInfo.Type.Equals(fieldInfo.FieldType));
+                Debug.Assert(parent == null || parent.FieldInfo.FieldType.Equals(fieldInfo.DeclaringType));
+
+                DeclaringTypeAndInfo = declaringTypeAndInfo;
+                FieldTypeAndInfo = fieldTypeAndInfo;
+                FieldInfo = fieldInfo;
+                Name = name;
+                Parent = parent;
+            }
+        }
+
+        private ReadOnlyCollection<Field> GetFields()
+        {
+            if (_lazyFields == null)
+            {
+                _lazyFields = GetFields(_typeAndInfo, _cardinality);
+            }
+            return _lazyFields;
+        }
+
+        private static ReadOnlyCollection<Field> GetFields(TypeAndCustomInfo declaringTypeAndInfo, int cardinality)
+        {
+            Debug.Assert(declaringTypeAndInfo.Type.GetTupleCardinalityIfAny() == cardinality);
+
+            var appDomain = declaringTypeAndInfo.ClrType.AppDomain;
+
+            var customTypeInfoMap = CustomTypeInfoTypeArgumentMap.Create(declaringTypeAndInfo);
+            var tupleElementNames = customTypeInfoMap.TupleElementNames;
+
+            var builder = ArrayBuilder<Field>.GetInstance();
+            Field parent = null;
+            int offset = 0;
+
+            while (true)
+            {
+                var declaringType = declaringTypeAndInfo.Type;
+                int n = Math.Min(cardinality, TypeHelpers.TupleFieldRestPosition - 1);
+                for (int index = 0; index < n; index++)
+                {
+                    var field = declaringType.GetTupleField(TypeHelpers.GetTupleFieldName(index));
```

Comment:
Hmm actually scratch that.  The `index` is correct here beause it's looking within this tuple type.  The `offset + index` below is also correct because it's synthesizing the user visible name. 

---
In reply to: [78263129](https://github.com/dotnet/roslyn/pull/13709#discussion_r78263129) [](ancestors = 78263129)